how-it-works/1_protocol/01_peer-to-peer-p2p.subpage.md

```diff
@@ -5,10 +5,73 @@ shareImage: /img/how-it-works/peer-to-peer-p2p.600.jpg
+Artifacts are the network messages that are to be broadcast in the subnet, such as ingress messages to canister smart contracts submitted by users or protocol-originating messages such as signature shares computed by the consensus layer of subnet nodes.
+P2P realizes a peer-to-peer broadcast network that guarantees the secure eventual broadcast delivery of an artifact.
+The P2P layer thereby is the communication fabric for the IC protocol stack and is used by the consensus layer, the next layer in the stack above it, to broadcast artifacts to the nodes in the subnet.
+
+## Gossip Protocol
+
+The P2P layer builds upon the basic principle of *gossip*.
+Gossip in communication networks works along the same basic principle as gossip among people: A node in the subnet is connected with a subset of the other nodes of the subnet – its *peers*.
+Whenever a node receives a message from a peer or creates one itself as part of the IC protocol, it gossips this message to all its peers.
+By every node in the network doing exactly this, every artifact *eventually* propagates through the whole subnet.
+
+## Asynchronous Communication Networks
+
+It is important to note that artifacts reach all subnet nodes eventually, that is, no upper bound on when this happens can be given.
+This important property of an *asynchronous communication network*, which reflects the real world of networks, is crucial and the IC consensus protocol is built with this assumption.
+Clearly, assuming an asynchronous network makes protocols building on top more complex than they would be when assuming a synchronous network.
+
+## Adverts
+
+Ensuring the eventual delivery of broadcasts to all nodes in the subnet is achieved by a node providing a received or protocol-generated message to all its connected peers.
+However, doing so in a naïve way by simply forwarding the message to all peers would result in nodes receiving as many copies of the same message as the node has peers, which unnecessarily consumes networking bandwidth for transmitting duplicate messages and reduces the achieveable throughput of the subnet.
+
+This duplication of delivered artifacts in the naïve approach is mitigated by nodes sending *adverts* for artifacts to their peers instead of sending the artifacts themselves.
+An advert specifies its corresponding artifact, but is a small message only containing the hash of the artifact to unambiguously refer to it and some additional metadata.
+Only adverts for artifacts are sent (pushed) to all its peers by a node.
+A node receiving adverts then requests each artifact only once from one of its peers that has advertised the artifact through an advert.
+For artifacts that are larger than the advert size, this saves bandwidth on the network connections between the peers – the savings grow with increasing size of the artifacts.
```

Comment:
I'd remove this sentence. We indeed don't have a mechanism to send small artifacts immediately, and this sentence just raises the question why without answering. The answer is that it's buried somewhere in the backlog and the gain is not big enough to invest the required time :)